### PR TITLE
test(cypress): add amount_captured coverage for payments manual update

### LINKED ISSUE #14226: [QA] Add test cases for amount_captured on Payments Manual Update API (PR #14136)
### Description

Add Cypress test coverage for the `amount_captured` field introduced on the Payments Manual Update API by PR #14136 (`feat(core): Add amount_captured field to manual update api`, merged to main).

`PUT /payments/{payment_id}/manual-update` now accepts an `amount_captured` field (partial capture) or `update_amount_captured: true` (capture the full remaining amount); the response returns `amount_capturable` and `amount_captured`.

### Coverage added

4 new test contexts in `cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js`:

| Scenario | Type | Expected |
|---|---|---|
| Manual update with `amount_captured` (partial capture, 2500 of 6000) | happy path | PUT 200 `{attempt_status: charged, amount_captured: 2500, amount_capturable: 6000}`; GET payment `status=succeeded`, `amount_received=2500` |
| Manual update with `update_amount_captured=true` (full capture) | happy path | PUT 200 `{attempt_status: charged, amount_captured: 6000}`; GET payment `status=succeeded`, `amount_received=6000` |
| `amount_captured` exceeding payment amount (6001 > 6000) | negative | PUT 422 `IR_06` — "amount_captured should be less than or equal to amount" |
| `amount_captured` together with `update_amount_captured=true` | edge case | PUT 422 `IR_06` — "amount_captured cannot be provided when update_amount_captured is true" |

Supporting changes:

- `cypress-tests/cypress/e2e/configs/Payment/Commons.js` — 4 new config keys: `ManualPaymentUpdateAmountCaptured`, `ManualPaymentUpdateUpdateAmountCaptured`, `ManualPaymentUpdateAmountCapturedExceedsAmount`, `ManualPaymentUpdateAmountConflict`
- `cypress-tests/cypress/support/commands.js` — extended `manualPaymentStatusUpdateTest` (`amount_captured`/`update_amount_captured` request passthrough, `amount_captured`/`amount_capturable` response assertions, expected-error branch for configured non-200 responses) and `retrievePaymentCallTest` (`expectedAmountReceived` assertion)

Related: #14136

### PR DESCRIPTION
**Pull request by HS-QUALIBOT**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Adds Cypress test coverage for the `amount_captured` field on the Payments Manual Update API (`PUT /payments/{payment_id}/manual-update`) introduced in #14136. Four new test contexts are added to the existing `47-ManualPaymentUpdate.cy.js` spec: two happy paths (manual update with a partial `amount_captured`, and with `update_amount_captured: true` for a full capture — each verified via a subsequent payment retrieve asserting persistence), one negative case (`amount_captured` greater than the payment amount → 422 `IR_06`), and one edge case (`amount_captured` supplied together with `update_amount_captured: true` → 422 `IR_06`). The flow is connector-agnostic and is exercised through the `stripe` connector in the regression suite.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

None of the above — this PR only touches the Cypress test suite:

- `cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js` — 4 new `amount_captured` contexts (happy path ×2, negative, edge case)
- `cypress-tests/cypress/e2e/configs/Payment/Commons.js` — 4 new `getCustomExchange` config keys: `ManualPaymentUpdateAmountCaptured`, `ManualPaymentUpdateUpdateAmountCaptured`, `ManualPaymentUpdateAmountCapturedExceedsAmount`, `ManualPaymentUpdateAmountConflict`
- `cypress-tests/cypress/support/commands.js` — extended `manualPaymentStatusUpdateTest` (`amount_captured`/`update_amount_captured` request passthrough, `amount_captured`/`amount_capturable` response assertions, expected-error branch for configured non-200 responses) and `retrievePaymentCallTest` (new `expectedAmountReceived` assertion)

## Motivation and Context
#14136 (`feat(core): Add amount_captured field to manual update api`, merged) shipped without Cypress coverage for the new field — the existing `47-ManualPaymentUpdate.cy.js` spec only covered status-only updates, invalid attempt IDs, and custom error persistence. This PR closes that coverage gap so regressions in partial/full capture amounts on the manual update API are caught by the test suite.

Closes #14226
Related to #14136

## How did you test it?
The full regression suite (changed spec `47-ManualPaymentUpdate.cy.js` plus its prerequisite specs) was executed against the `stripe` connector in the QA pipeline worktree. All 14 tests passed. The `RUNNER_RESULT` block from the QA pipeline is included verbatim below.

### Changed-spec verification — `stripe`

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  ExecutedFrom: /Users/likhin.bopanna/Desktop/cypress-tests-CLA-307/cypress-tests (worktree qa/CLA-307 — NOT main repo)
  WorktreeSpecVerification: grep -c "it(" = 8 (correct worktree copy confirmed before run)
  PrereqsUsed: spec/Payment/01-AccountCreate.cy.js, 02-CustomerCreate.cy.js, 03-ConnectorCreate.cy.js
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  PerSpecResults:
    - 01-AccountCreate.cy.js: 3 tests, 3 passed, 0 failed, 0 skipped
    - 02-CustomerCreate.cy.js: 1 test, 1 passed, 0 failed, 0 skipped
    - 03-ConnectorCreate.cy.js: 2 tests, 2 passed, 0 failed, 0 skipped
    - 47-ManualPaymentUpdate.cy.js: 8 tests, 8 passed, 0 failed, 0 skipped

  Spec47AllTestTitles:
    1. [PASS] Create Payment Intent -> Manual Update -> Retrieve Payment
    2. [PASS] Create Payment Intent -> Manual Update Status Only -> Retrieve Payment
    3. [PASS] Create Payment Intent -> Manual Update with amount_captured -> Retrieve Payment  [NEW amount_captured happy path, 2500]
    4. [PASS] Create Payment Intent -> Manual Update with update_amount_captured -> Retrieve Payment  [NEW update_amount_captured=true happy path]
    5. [PASS] Create Payment Intent -> Manual Update with Invalid Attempt ID
    6. [PASS] Create Payment Intent -> Manual Update with amount_captured exceeding payment amount  [NEW negative — 422 IR_06]
    7. [PASS] Create Payment Intent -> Manual Update with amount_captured and update_amount_captured conflict  [NEW negative — 422 IR_06]
    8. [PASS] Create Payment Intent -> Manual Update with Custom Error -> Verify Persistence

  All 4 NEW amount_captured contexts confirmed present and passing (items 3, 4, 6, 7).

  Failures: []
  SkippedTests: []
  FlakeyTests: []
  BlockedReasons: []
```

### QA gate

```
GATE_PASSED:
  ConnectorRegressions:
    - Connector: stripe
      Result: PASS
  AllChecksPassed: YES
  ReadyForGitHubAgent: YES
```

### Post-review-fix verification — `stripe`

Re-ran the full suite after the review-fix revision (`ca4afb7f11` — retrieve assertions now driven from the Commons.js config keys). All 14 tests passed. Run executed from the worktree `/Users/likhin.bopanna/Desktop/cypress-tests-CLA-307/cypress-tests` (branch `qa/CLA-307`), spec 47 verified at 8 `it` blocks before running.

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  Failures:
    none

  SkippedTests:
    none

  FlakeyTests:
    none

  BlockedReasons:
    none
```

### Summary

| Connector | Specs Run | Passed | Failed | Skipped | Status |
|---|---|---|---|---|---|
| stripe | 4 | 14 | 0 | 0 | PASS |

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

## Post-review-fix verification (Round 2)

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  Per-Spec Breakdown:
    - 01-AccountCreate.cy.js:  3/3 passing
    - 02-CustomerCreate.cy.js:  1/1 passing
    - 03-ConnectorCreate.cy.js: 2/2 passing
    - 47-ManualPaymentUpdate.cy.js: 8/8 passing

  Failures:
    none

  SkippedTests:
    none

  FlakeyTests:
    none

  BlockedReasons:
    none
```

## Post-review-fix verification (Round 3)

Round 3 revision (spec-side retrieve assertions removed, verification driven from the configs layer) re-run on stripe: 14/14 passing (01-AccountCreate 3/3, 02-CustomerCreate 1/1, 03-ConnectorCreate 2/2, 47-ManualPaymentUpdate 8/8). Mochawesome report: `cypress/reports/stripe/stripe_report_004.html`.

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  Failures: none

  SkippedTests: none

  FlakeyTests: none

  BlockedReasons: none
```

### CODE REVIEW

File: cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js [hunk 1]
```diff
@@ -155,6 +155,138 @@ describe("Payment Manual Update Tests", () => {
...
+        cy.retrievePaymentCallTest({
+          globalState,
+          data: {
+            Configs: {
+              skipBillingAssertion: true,
+            },
+          },
+          unconfirmedPayment: true,
+          expectedIntentStatus: "succeeded",
+          expectedAmountReceived: 2500,
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js]: expectedAmountReceived and expectedIntentStatus will be verified in the commons.js(configs ) , its not necessary to verify here .

File: cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js [hunk 2]
```diff
@@ -155,6 +155,138 @@ describe("Payment Manual Update Tests", () => {
...
+        }
+
+        cy.retrievePaymentCallTest({
+          globalState,
+          data: {
+            Configs: {
+              skipBillingAssertion: true,
+            },
+          },
+          unconfirmedPayment: true,
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js:280]: expectedAmountReceived and expectedIntentStatus will be verified in the commons.js(configs ) , its not necessary to verify here .

File: cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js (hunk 1 above)
Comment (HS-QUALIBOT) [reply cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js]: Addressed in ca4afb7f11 — the retrieve assertions are now fully config-driven from Commons.js:

- `ManualPaymentUpdateAmountCaptured` now carries `Configs.skipBillingAssertion`, `Response.body.amount_captured: 2500`, and `Response.body.status: "succeeded"`
- `retrievePaymentCallTest` derives the `amount_received` / `status` expectations from the connector config (`data.Response.body`), so the spec passes zero inline expected values
- The dead `expectedAmountReceived` option was removed from the command signature

Suite re-verified green after the change:

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  Failures:
    none

  SkippedTests:
    none

  FlakeyTests:
    none

  BlockedReasons:
    none
```

File: cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js (hunk 2 above)
Comment (HS-QUALIBOT) [reply cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js:280]: Addressed in ca4afb7f11 — the retrieve assertions are now fully config-driven from Commons.js:

- `ManualPaymentUpdateUpdateAmountCaptured` now carries `Configs.skipBillingAssertion`, `Response.body.amount_captured: 6000`, and `Response.body.status: "succeeded"`
- `retrievePaymentCallTest` derives the `amount_received` / `status` expectations from the connector config (`data.Response.body`), so the spec passes zero inline expected values
- The dead `expectedAmountReceived` option was removed from the command signature

Suite re-verified green after the change:

```
RUNNER_RESULT:
  Connector: stripe
  SpecFile: cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 14
  Passed: 14
  Failed: 0
  Skipped: 0
  OverallStatus: PASS

  Failures:
    none

  SkippedTests:
    none

  FlakeyTests:
    none

  BlockedReasons:
    none
```

File: cypress-tests/cypress/support/commands.js [hunk 3]
```diff
@@ -4906,6 +4907,25 @@ Cypress.Commands.add(
             ).to.equal(expectedIntentStatus);
           }
 
+          // Retrieve expectations for manual-update flows, driven by the
+          // connector config (Commons.js). Scoped to config keys that define
+          // `amount_captured` in their Response body (only the manual-update
+          // keys do) so flow-step keys (Capture / Confirm / ...) passed by
+          // other specs are unaffected.
+          if (typeof resData.amount_captured !== "undefined") {
```
Comment (likhinbopanna) [inline cypress-tests/cypress/support/commands.js]: Why this is required , we can verify the amount_captured in the configs.js file only

Comment (HS-QUALIBOT) [reply cypress-tests/cypress/support/commands.js]: Addressed in 1fdf44089a — the bespoke block is removed; `retrievePaymentCallTest` is now byte-identical with main (no manual-update branching in any shared command).

The `amount_captured` verification now lives entirely in the configs layer: the `Commons.js` `card_pm` keys carry the expected `amount_captured` (2500/6000) and `status` values; the spec resolves them via `getConnectorDetails` and asserts `response.body.amount_received` equals `data.Response.body.amount_captured` in a chained `.then()`, following the existing pattern in `51-ExtendedCardInfo.cy.js` (lines 211-221). Status is verified via the command's pre-existing `expectedIntentStatus` param.

Verified on stripe: 14/14 passing (including all 8 spec-47 tests), 0 failures.

File: cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js [hunk 4]
```diff
@@ -155,6 +155,146 @@ describe("Payment Manual Update Tests", () => {
...
+
+        const data = getConnectorDetails(globalState.get("connectorId"))[
+          "card_pm"
+        ]["ManualPaymentUpdateUpdateAmountCaptured"];
+
+        cy.retrievePaymentCallTest({
+          globalState,
+          data,
+          unconfirmedPayment: true,
+          expectedIntentStatus: data.Response.body.status,
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js]: status should be verified in Commons.js file and these assertions are not required

Comment (HS-QUALIBOT) [reply cypress-tests/cypress/e2e/spec/Payment/47-ManualPaymentUpdate.cy.js]: Addressed in `d676805271`.

- **Spec-side assertions removed**: both amount-related Retrieve steps in `47-ManualPaymentUpdate.cy.js` are now plain config-driven retrieves — `cy.retrievePaymentCallTest({ globalState, data, unconfirmedPayment: true })` — with zero assertion lines, matching the repo convention for post-mutation retrieves (cf. `06-NoThreeDSManualCapture.cy.js` "Retrieve Payment after Partial Capture").
- **Verification lives in the configs layer**: the `Commons.js` `card_pm` `ManualPaymentUpdate*` keys carry the feature contract (`attempt_status`, `amount_captured` 2500/6000, `amount_capturable`), which `manualPaymentStatusUpdateTest` asserts against the live `PUT /payments/{id}/manual-update` response; the negative 422 cases (IR_06) are verified from the same config error contracts. The unconsumed `status` field was removed from those keys.
- `retrievePaymentCallTest` remains byte-identical with main.
- Verified on stripe: 14/14 passing (including all 8 spec-47 tests), 0 failures.

**Review by venkatakarthikm-maker — APPROVED** [review]

File: cypress-tests/cypress/e2e/configs/Payment/Commons.js [hunk 5]
```diff
@@ -4029,6 +4029,74 @@ export const connectorDetails = {
...
+        },
+      },
+    }),
+    ManualPaymentUpdateUpdateAmountCaptured: getCustomExchange({
+      Configs: {
+        skipBillingAssertion: true,
+      },
+      Request: {
+        attempt_status: "charged",
+        update_amount_captured: true,
```
Comment (Gnanasundari24) [inline cypress-tests/cypress/e2e/configs/Payment/Commons.js:4089]: If we pass update_amount_captured: true, what is the exact behaviour?
